crates/deltalake-core/src/operations/write.rs

```diff
@@ -363,6 +381,133 @@ pub(crate) async fn write_execution_plan(
         .collect::<Vec<_>>())
 }
 
+#[allow(clippy::too_many_arguments)]
+pub(crate) async fn write_execution_plan(
+    snapshot: Option<&DeltaTableState>,
+    state: SessionState,
+    plan: Arc<dyn ExecutionPlan>,
+    partition_columns: Vec<String>,
+    object_store: ObjectStoreRef,
+    target_file_size: Option<usize>,
+    write_batch_size: Option<usize>,
+    writer_properties: Option<WriterProperties>,
+    safe_cast: bool,
```

Comment:
I agree but perhaps it deserves a dedicated issues because `write_execution_plan` is used by many operations and probably not a good idea to make it part of this PR. I'll create a new issue for this